c7n/manager.py

```diff
@@ -91,10 +91,15 @@ def get_resource_manager(self, resource_type, data=None):
         if klass is None:
             raise ValueError(resource_type)
 
-        # if we're already querying via config carry it forward
-        if not data and self.source_type == 'config' and getattr(
-                klass.get_model(), 'config_type', None):
-            return klass(self.ctx, {'source': self.source_type})
+        # propagate source by default
+        if data is None:
```

Comment:
previously automatic source propagation only worked for AWS Config sources, this enables it to work by default across other provider sources.